src/main/java/org/bsoftware/ward/services/implementation/InfoService.java

```diff
@@ -72,7 +72,12 @@ private void getMachineInfo()
         infoDto.setOperatingSystemInfo(operatingSystem.getFamily() + " " + osVersionInfo.getVersion() + ", " + osVersionInfo.getCodeName());
 
         infoDto.setTotalRam(converterUtility.getConvertedCapacity(globalMemory.getTotal()) + " Ram");
-        infoDto.setRamType(globalMemory.getPhysicalMemory().get(0).getMemoryType());
+        List<PhysicalMemory> physicalMem = globalMemory.getPhysicalMemory();
```

Comment:
I'd have used this instead:
```java
final String ramType = physicalMem.stream()
    .findFirst()
    .map(PhysicalMemory::getMemoryType)
    .orElse("Unknown");
infoDto.setRamType(ramType);

No more checking for emptyness and the code is self-explanatory with the default value when it doesn't find something.

Reply:
Actually, it should be `.findAny()`, which doesn't change the actual execution, it's more about semantics(i.e. pick any memory you can find instead of first).